src/librustc_mir/transform/const_prop.rs

```diff
@@ -52,7 +52,7 @@ struct OptimizationFinder<'b, 'a, 'tcx:'a+'b> {
     mir: &'b Mir<'tcx>,
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
     source: MirSource,
-    optimizations: OptimizationList<'tcx>,
+    places: FxHashMap<Local, Const<'tcx>>,
```

Comment:
Please use `IndexVec<Local, Option<Const<'tcx>>>`.
Also, rename `OptimizationFinder` to `ConstPropagator` or `ConstFolder`.

Reply:
Remove comments about "optimization opportunities" and add this comment:
```rust
// FIXME(oli-obk, eddyb) Optimize locals (or even local paths) to hold
// constants, instead of just checking for const-folding succeeding.
// That would require an uniform one-def no-mutation analysis
// and RPO (or recursing when needing the value of a local).
```